File: XFile13.cpp
#include "pt4.h"
#include <fstream>
using namespace std;

void Solve()
{
    Task("XFile13");
    string s1, s2 , s3;
	pt >> s1 >> s2 >> s3;
	ifstream f1(s1, ios::binary);
	ofstream f2(s2, ios::binary);
	ofstream f3(s3, ios::binary);
	f1.seekg(-4, ios::end);
	while (f1.peek() != -1)
	{
		int x;
		f1.seekg(f1.tellg());
		f1.read((char *)&x, sizeof(x));
		f1.seekg(-8, ios::cur);
		if(x > 0)
		f2.write((char *)&x, sizeof(x));
		if(x < 0)
		f3.write((char *)&x, sizeof(x));
	}
	f1.close();
	f2.close();
	f3.close();
}

File: XFile13.cs
// File: "XFile13"
using PT4;
using System.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static async void Solve()
        {
            Task("XFile13");
            FileStream f = new FileStream(GetString(), FileMode.Open, FileAccess.ReadWrite);
            FileStream f1 = new FileStream(GetString(), FileMode.OpenOrCreate, FileAccess.ReadWrite);
            FileStream f2 = new FileStream(GetString(), FileMode.OpenOrCreate, FileAccess.ReadWrite);
            for (int i = 1; i <= f.Length / 4; i++)
            {
                byte[] b = new Byte[4];
                f.Seek(-4 * i, SeekOrigin.End);
                f.Read(b, 0, 4);
                if (System.BitConverter.ToInt16(b, 0) > 0)
                    f1.Write(b, 0, 4);
                else
                    f2.Write(b, 0, 4);

            }
            f1.Close();
            f2.Close();

        }
    }
}
